botorch/models/transforms/input.py

```diff
@@ -340,6 +343,21 @@ def __init__(
                 zero errors.
         """
         super().__init__()
+        if (indices is None) or (len(indices) == 0):
+            self.register_buffer(
+                "indices", torch.tensor(list(range(d)), dtype=torch.long)
+            )
```

Comment:
I wonder if a slightly better design would be to just not register an `indices` buffer and then then we just gate the indexing in the `_transform` and `_untransform` methods by sth like `if hasattr(self, "indices")):`. That would have the advantage of not doing a nullop clone + indexing operation in the all-indices case, which I anticipate to be the most common one.

Reply:
Good, idea. I will change it accordingly.